spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java

```diff
@@ -167,8 +170,10 @@ else if(this.applicationFailedEvent != null){
 			if(this.closeContext && this.context.isActive()) {
```

Comment:
We need to move the `this.finished = true;` to above where we close the context.  Else the onDestroy fires prior  to the finished being set to true.  I can do this on merge if you like.